ibm/service/dnsservices/resource_ibm_private_dns_custom_resolver_secondary_zone.go

```diff
@@ -0,0 +1,269 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package dnsservices
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	pdnsSecondaryZoneID           = "secondary_zone_id"
+	pdnsSecondaryZoneZone         = "zone"
+	pdnsSecondaryZoneTransferFrom = "transfer_from"
+	pdnsSecondaryZoneEnabled      = "enabled"
+	pdnsSecondaryZoneDescription  = "description"
+	pdnsSecondaryZoneCreatedOn    = "created_on"
+	pdnsSecondaryZoneModifiedOn   = "modified_on"
+)
+
+func ResourceIBMPrivateDNSSecondaryZone() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMPrivateDNSSecondaryZoneCreate,
+		ReadContext:   resourceIBMPrivateDNSSecondaryZoneRead,
+		UpdateContext: resourceIBMPrivateDNSSecondaryZoneUpdate,
+		DeleteContext: resourceIBMPrivateDNSSecondaryZoneDelete,
+		Exists:        resourceIBMPrivateDNSSecondaryZoneExists,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			pdnsInstanceID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The unique identifier of a service instance.",
+			},
+			pdnsResolverID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The unique identifier of a custom resolver.",
+			},
+			pdnsSecondaryZoneID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Secondary Zone ID",
+			},
+
+			pdnsSecondaryZoneZone: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Secondary Zone Zone",
```

Comment:
Schema descriptions to match with updated description from the docs [ Can be seen in api docs ]